### PR TITLE
Consolidate sapling functionality in zcash_client_sqlite in a dedicated module.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: zcash_client_sqlite/src/lib.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ use zcash_primitives::{
     consensus::{self, BlockHeight},
     legacy::TransparentAddress,
     memo::{Memo, MemoBytes},
-    sapling::{self, Nullifier},
+    sapling::{self},
```
Comment (daira) [inline zcash_client_sqlite/src/lib.rs:46]: Is this `rustfmt`ted?

File: zcash_client_sqlite/src/wallet/sapling.rs [hunk 2]
```diff
@@ -173,6 +228,182 @@ pub(crate) fn select_spendable_sapling_notes<P>(
...
+            rcm,
+            nf,
+            memo,
+            is_change,
+        )
+    } else {
+        // It was there, so grab its row number.
+        stmts.stmt_select_received_note(tx_ref, output.index())
+    }
+}
```
Comment (daira) [inline zcash_client_sqlite/src/wallet/sapling.rs:405]: I have not checked that the code here has only been moved.

**Review by daira — APPROVED** [review]
utACK, modulo not checking the changes that should be move-only.

File: zcash_client_sqlite/src/lib.rs (hunk 1 above)
Comment (nuttycom) [reply zcash_client_sqlite/src/lib.rs:46]: For some reason rustfmt doesn't collapse this to just `sapling`. This code is likely to change further, so I'm going to leave it like this for now.

**Pull request merged**